app/controllers/concerns/authentication_and_sso_concerns.rb

```diff
@@ -112,6 +112,24 @@ def set_sso_cookie!
     }
   end
 
+  def set_sso_saml_cookie!
+    return unless Settings.sso.cookie_enabled &&
+                  @session_object.present? &&
+                  # if the user logged in via SSOe, there is no benefit from
+                  # creating a MHV SSO shared cookie
+                  !@current_user&.authenticated_by_ssoe
+
+    Rails.logger.info('SSO: ApplicationController#set_sso_cookie!', sso_logging_info)
+
+    cookies[Settings.sso.saml_cookie_name] = {
+      value: sso_saml_cookie_content,
+      expires: nil, # NOTE: we track expiration as an attribute in "value." nil here means kill cookie on browser close.
+      secure: Settings.sso.cookie_secure,
+      httponly: true,
+      domain: Settings.sso.cookie_domain
+    }
+  end
```

Comment:
Looking good! To improve test-ability, perhaps consider the following refactor:

```
def set_sso_saml_cookie!
    SSO::SAMLCookie.from(@session_object, @current_user)
end
```

and then in another file (maybe in lib?)

```
module SSO
    class SAMLCookie
        def self.from(session_object, current_user)
            # …{refactored from set_sso_saml_cookie}
        end

        def self.cookie_content
            …{refactored from sso_saml_cookie_content}
        end
    end
end
```